oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/expression/NumberMatch.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.skywalking.oap.server.core.analysis.metrics.expression;
+
+import org.apache.skywalking.oap.server.core.analysis.metrics.annotation.FilterMatcher;
+
+@FilterMatcher
+public class NumberMatch {
+
+    public boolean match(int left, int right) {
+        return left == right;
+    }
+
+    public boolean match(long left, long right) {
+        return left == right;
+    }
+
+    public boolean match(float left, float right) {
+        return left == right;
+    }
+
+    public boolean match(Number left, Number right) {
+        return left.equals(right);
+    }
```

Comment:
```suggestion
```

Number doesn't include float like I mentioned.
```
numberConditionValue
    : NUMBER_LITERAL
    ;

NUMBER_LITERAL :   Digits+;

fragment Digits
    : [0-9] ([0-9_]* [0-9])?
    ;
```

And Number object doesn't make sense too.

Reply:
@YczYanchengzhe I think supporting `long / Long`, `int / Integer` should be enough in our cases, we only have 1 `double` typed field in `Source` scope and its name is `usePercent`, I don't think it makes sense to compare `==` between `double`s

Reply:
I agree